### ISSUE TITLE
Question - Block and Coordinates

### ISSUE THREAD

**Issue by ikwtif** [mozman/ezdxf#123]
Not sure If I'm missing something about Blocks, but I am not able to find it so thought to just ask the question here. What I want to do is check locations from objects (lines, text) which are inside a block and compare them to other objects that are not in blocks. So I wonder if it is possible to do one of these following points

- get coordinates of objects (lines, text) from inside a block based on the model-space coordinates

- get the coordinates of the actual block in model-space coordinates? If this is possible I could just get the coordinates for the objects inside the block-layout and add them to the block coordinates to get their actual position in the modelspace.

**Comment by ikwtif** [comment]
Strange how when I post the question there is a high chance I find the answer soon afterwards
I'll leave the question up for now, as maybe point 1 is still possible, although I doubt it. But I use this code now for getting the block insert. Maybe this can be added to the documentation somewhere?

```
blockrefs = msp.query('INSERT')
if len(blockrefs):
    for entity in blockrefs:
        print(entity.dxf.insert)
```

**Comment by mozman** [comment]
Look at the documentation: https://ezdxf.mozman.at/docs/tutorials/blocks.html

Your first request "get coordinates of objects (lines, text) from inside a block based on the model-space coordinates" is not possible, it is more complex than you expect.

**Comment by mozman** [comment]
I have an idea how add some support for point 1 for simple cases, but this will take some time.

It should look like that:

```python
# create an UCS at the INSERT location, with rotation 
# and scaling support, which is not available yet
ucs = insert.ucs()

# iterate over BLOCK entities and transform the BLOCK 
# coordinates into WCS
for e in insert.block():
    if e.dxftype() == 'LINE':
         wcs_start = ucs.to_wcs(e.dxf.start)
         wcs_end = ucs.to_wcs(e.dxf.end)

```

**Comment by ikwtif** [comment]
Thanks for the response. I read the part from the docs with the difficulty about blocks, but wanted to double check as I wasn't sure I was overlooking something. And for some reason I thought I was able to do it once, but clearly that's not the case. 

Support for doing this would be nice, but I guess it' not a highly requested feature since you can just explode the blocks in advance, which I'll probably do for now. After I finish my program, I might look into maybe creating the feature myself at some point. Thank you for the explanation!

**Comment by mozman** [comment]
Released beta version 0.12b1 with (experimental) support for exploding block references and examining of "virtual" entities of block references, see blocks [tutoriual](https://ezdxf.mozman.at/docs/tutorials/blocks.html#examine-entities-of-block-references).
